### PR TITLE
Bug fixes in alignment routine

### PR DESCRIPTION
**Pull request by ehwenk**
* fixes to strip_names functions
* a few edits to match protocols (match_taxa.R)
* arrange APC names so `accepted` taxon names are chosen before `known` names

### CODE REVIEW

File: R/match_taxa.R [hunk 1]
```diff
@@ -59,10 +59,11 @@ match_taxa <- function(taxa, resources, dataset_id = "XXXX") {
   i <-
     (
       stringr::str_detect(taxa$tocheck$stripped_name, "[:space:]sp$") |
-        stringr::str_detect(taxa$tocheck$stripped_name, "[:space:]spp$")
+        stringr::str_detect(taxa$tocheck$stripped_name, "[:space:]spp$") |
+        stringr::str_detect(taxa$tocheck$stripped_name, "[:space:]ssp$")
```
Comment (dfalster) [inline R/match_taxa.R:63]: These two lines are identical. should one of them be stripped_name2?

File: R/match_taxa.R [hunk 2]
```diff
@@ -96,10 +97,11 @@ match_taxa <- function(taxa, resources, dataset_id = "XXXX") {
...
```
Comment (dfalster) [inline R/match_taxa.R:101]: as above

File: R/match_taxa.R [hunk 3]
```diff
@@ -131,10 +133,11 @@ match_taxa <- function(taxa, resources, dataset_id = "XXXX") {
...
```
Comment (dfalster) [inline R/match_taxa.R:137]: as above

File: R/utils.R [hunk 4]
```diff
@@ -19,9 +19,10 @@
 #' @noRd
 strip_names <- function(taxon_names) {
   taxon_names %>%
+    stringr::str_replace_all("\\.", "") %>%
     stringr::str_replace_all("[:punct:]", " ") %>%
-    stringr::str_replace_all(" subsp ", " ") %>%
```
Comment (dfalster) [inline R/utils.R:23]: is this deletion intentional

**Review by dfalster — DISMISSED** [review]
Looks good Lizzy, a few queries. Feel free to merge once these are addressed.

@fontikar I've reviewed so no need for you to

File: R/match_taxa.R (hunk 1 above)
Comment (ehwenk) [reply R/match_taxa.R:63]: One line is `ssp` and the other is `spp` - occasionally both abbreviations have occurred

File: R/match_taxa.R (hunk 2 above)
Comment (ehwenk) [reply R/match_taxa.R:101]: One line is `ssp` and the other is `spp` - occasionally both abbreviations have occurred

File: R/match_taxa.R (hunk 3 above)
Comment (ehwenk) [reply R/match_taxa.R:137]: One line is `ssp` and the other is `spp` - occasionally both abbreviations have occurred

File: R/utils.R (hunk 4 above)
Comment (ehwenk) [reply R/utils.R:23]: yes it was, but you forced me to look harder at a few pieces and what I've just changed means more names are aligned at earlier stages in the match cycle

**Review by fontikar — APPROVED** [review]

**Pull request merged**